### PR TITLE
Marshal Fixes and Storyteller Reporting

### PR DESCRIPTION
**Pull request by NanakoAC**
This PR Does three main things

1. Tweaks and fixes to the Ironhammer Marshal antitag
After discussion with clock about marshal's lore and gameplay role, i've written new greeting text explaining a bit about their role in the universe and responsibilities

Fixed marshal's preferences flag to Crew Sided as it ought to be. It was previously traitor and players reported this as a bug.
Made sure marshal can't spawn without at least one viable antag to be a target

2. Storyteller Fail Reporting
When the storyteller is set to debug mode, attempting to force spawn an antag will now explain failure conditions to the user if something goes wrong. This is useful both for debugging and for realtime information for admins.

3. Assassinate
Several uses of assassinate in the codebase used an inconsistent spelling, assasinate or asassinate. These have been mass replaced to the correct version which is "assassinate", with four S's

### CODE REVIEW

File: code/datums/objective/assasinate_marshal.dm [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
 	var/target = src.target ? "[src.target.current.real_name], the [src.target.assigned_role]" : "no_target"
-	return "Find and assassinate <a href='?src=\ref[src];switch_target=1'>[target]</a>."
+	return "You are after the fugitive <a href='?src=\ref[src];switch_target=1'>[target]</a>."
 
 /datum/objective/assassinate/marshal/get_info()
 	if(target)
 		return "(target was [target.name])"
 
-/datum/objective/assassinate/marshal/update_explanation()
+/datum/objective/assassinate/marshal/marshal/update_explanation()
```
Comment (LethalGhost) [inline code/datums/objective/assasinate_marshal.dm]: > assassinate/marshal/marshal

Isn't that a typo?

File: code/datums/objective/assasinate_marshal.dm [hunk 2]
```diff
@@ -1,30 +1,35 @@
...
-/datum/objective/assasinate/marshal/get_targets_list()
+/datum/objective/assassinate/marshal/get_targets_list()
 	var/list/possible_targets = list()
 	for(var/datum/mind/possible_target in SSticker.minds)
-		if(possible_target != owner && ishuman(possible_target.current) && (possible_target.current.stat != 2) && \
- 		player_is_ship_antag(possible_target))
+		if(possible_target != owner &&\
+		 ishuman(possible_target.current) &&\
+		  (possible_target.current.stat != 2) &&\
+		   (player_is_ship_antag(possible_target)))
```
Comment (LethalGhost) [inline code/datums/objective/assasinate_marshal.dm]: Is there something wrong with indents here?

File: code/game/antagonist/station/traitor.dm [hunk 3]
```diff
@@ -57,8 +57,8 @@
 	only_human = FALSE
 
 	possible_objectives = list(
-	/datum/objective/assasinate = 100,
-	/datum/objective/assasinate = 90,
+	/datum/objective/assassinate = 100,
+	/datum/objective/assassinate = 90,
```
Comment (LethalGhost) [inline code/game/antagonist/station/traitor.dm]: That's not target of PR, but on of lines above is redundant.

File: code/game/antagonist/antagonist_helpers.dm [hunk 4]
```diff
@@ -1,29 +1,42 @@
-/datum/antagonist/proc/can_become_antag(var/datum/mind/player)
+//If report is set, we will spam them with data about why the spawning failed
+/datum/antagonist/proc/can_become_antag(var/datum/mind/player, var/mob/report)
```
Comment (LethalGhost) [inline code/game/antagonist/antagonist_helpers.dm]: Shouldn't that be called like "report_to"?
I'm not sure.

File: code/datums/objective/assasinate_marshal.dm [hunk 5]
```diff
@@ -1,30 +1,35 @@
...
 
-/datum/objective/assasinate/marshal/get_targets_list()
+/datum/objective/assassinate/marshal/get_targets_list()
 	var/list/possible_targets = list()
 	for(var/datum/mind/possible_target in SSticker.minds)
-		if(possible_target != owner && ishuman(possible_target.current) && (possible_target.current.stat != 2) && \
- 		player_is_ship_antag(possible_target))
+		if(possible_target != owner &&\
+		 ishuman(possible_target.current) &&\
+		  (possible_target.current.stat != 2) &&\
```
Comment (LethalGhost) [inline code/datums/objective/assasinate_marshal.dm]:
```suggestion
		  (possible_target.current.stat != DEAD) &&\
```

File: code/datums/objective/assasinate_marshal.dm [hunk 6]
```diff
@@ -1,30 +1,35 @@
 /datum/objective/assassinate/marshal
 
+
 /datum/objective/assassinate/marshal/get_panel_entry()
 	var/target = src.target ? "[src.target.current.real_name], the [src.target.assigned_role]" : "no_target"
-	return "Find and assassinate <a href='?src=\ref[src];switch_target=1'>[target]</a>."
+	return "You are after the fugitive <a href='?src=\ref[src];switch_target=1'>[target]</a>."
```
Comment (LethalGhost) [inline code/datums/objective/assasinate_marshal.dm]: `panel_entry` shows up in antag panel (only visible for admins). "Find and assassinate" seems like more obvious description.

File: code/game/gamemodes/roleset/simple.dm [hunk 7]
```diff
@@ -50,12 +50,28 @@
 	req_crew = 10
 	event_pools = list(EVENT_LEVEL_ROLESET = -30) //This is an antitag, it has a negative cost to allow more antags to exist
 
+/datum/storyevent/roleset/marshal/can_trigger(var/severity, var/report)
+	var/a_count = 0
+	for(var/datum/antagonist/A in current_antags)
+		if(A.owner && A.is_active() && !A.is_dead())
```
Comment (LethalGhost) [inline code/game/gamemodes/roleset/simple.dm]: `is_active()` include `is_dead()` check.
And probably should include `owner` check (but can missed it right now)

File: code/game/gamemodes/storyteller.dm [hunk 8]
```diff
@@ -305,7 +305,10 @@ var/datum/storyteller/storyteller = null
 /*Sets up an event to be fired in the near future. This keeps things unpredictable
 The actual fire event proc is located in storyteller_meta*/
 /datum/storyteller/proc/schedule_event(var/datum/storyevent/C, var/event_type)
-	var/handle = addtimer(CALLBACK(GLOBAL_PROC, .proc/fire_event, C, event_type), rand(1, event_schedule_delay), TIMER_STOPPABLE)
+	var/delay = rand(1, event_schedule_delay)
+	if (event_type == EVENT_LEVEL_ROLESET)
+		delay = 1 //Basically no delay on these to reduce bugginess
```
Comment (LethalGhost) [inline code/game/gamemodes/storyteller.dm]: May be it's worth it to set default delay to 1 and call `rand` only if `event_type != EVENT_LEVEL_ROLESET`?

File: code/game/antagonist/station/traitor.dm (hunk 3 above)
Comment (NanakoAC) [reply code/game/antagonist/station/traitor.dm]: I'm not 100% certain, but i think that's a 90% chance of getting a second target in addition to the first

File: code/datums/objective/assasinate_marshal.dm (hunk 6 above)
Comment (NanakoAC) [reply code/datums/objective/assasinate_marshal.dm]: Asassinate is too...evil. 
The marshal is supposed to be one of the good guys, or at least neutral. I'd like to make capturing and jailing the target alive count too

File: code/datums/objective/assasinate_marshal.dm (hunk 2 above)
Comment (NanakoAC) [reply code/datums/objective/assasinate_marshal.dm]: these are continued lines, note the \
thats really all one line of code

File: code/game/antagonist/station/traitor.dm (hunk 3 above)
Comment (NanakoAC) [reply code/game/antagonist/station/traitor.dm]: no you're right it is redundant, fixing

**Pull request merged**